launcher/net/Download.cpp

```diff
@@ -97,8 +95,11 @@ void Download::startImpl()
     }
 
     request.setHeader(QNetworkRequest::UserAgentHeader, BuildConfig.USER_AGENT);
+    if (request.url().host().contains("api.curseforge.com")) {
```

Comment:
I think it would be better if you subclassed `Download` and did this change in there. It's no cool to check this for every download we want to do

Reply:
it doesn't cause much overhead anyways, just a blanket scan for the domain (but could pollute the generic nature of the `Download` class)

Reply:
You can just create a subclass like you would normally do, but override `startImpl()` to be what's now the `Download`'s one. Then use it instead of the common `Download` task when doing Curseforge stuff.

Maybe, instead of doing this, you can also create a `setHeaderEntry(QString key, QString value)` virtual method in `Download`, and call it on Curseforge-related downloads.

If you don't want to however, it's fine as it is, I'll have to fix the conflicts in my PR either way, so I can do this there :slightly_smiling_face:

Reply:
🥴 I'm a bit burnt out right now

I'll leave you to do it @flowln 💩